pkg/config/decode_hooks.go

```diff
@@ -0,0 +1,75 @@
+package config
+
+import (
+	"os"
+	"reflect"
+	"regexp"
+
+	"github.com/go-errors/errors"
+)
+
+var envPattern = regexp.MustCompile(`^env\((.*)\)$`)
+
+// LoadEnvHook is a mapstructure decode hook that loads environment variables
+// from strings formatted as env(VAR_NAME).
+func LoadEnvHook(f reflect.Kind, t reflect.Kind, data interface{}) (interface{}, error) {
+	if f != reflect.String {
+		return data, nil
+	}
+	value := data.(string)
+	if matches := envPattern.FindStringSubmatch(value); len(matches) > 1 {
+		if env := os.Getenv(matches[1]); len(env) > 0 {
+			value = env
+		}
+	}
+	return value, nil
+}
+
+// ValidateFunctionsHookFunc is a mapstructure decode hook that validates the functions config format.
+func ValidateFunctionsHookFunc(f reflect.Type, t reflect.Type, data interface{}) (interface{}, error) {
```

Comment:
```suggestion
// ValidateFunctionsHook is a mapstructure decode hook that validates the functions config format.
func ValidateFunctionsHook(f reflect.Type, t reflect.Type, data interface{}) (interface{}, error) {
```

nitpick: a `HookFunc` is a function that returns a hook. Subtle difference but I think it's worth having consistent names.